src/pages/recipes/index.tsx

```diff
@@ -0,0 +1,272 @@
+import { Layout } from "@/components/Layout/Layout";
+import { DEFAULT_PAGE_SIZE } from "@/constants";
+import { ErrorResponse, RecipesResponse } from "@/types";
+import fetcher from "@/lib/fetcher";
+import {
+  Box,
+  DataTable,
+  Divider,
+  Link,
+  PageHeader,
+  Pagination,
+  Section,
+  Typography,
+} from "@aivenio/aquarium";
+import { useEffect, useState } from "react";
+import useSWR from "swr";
+import { EmptyState } from "@/components/EmptyState/EmptyState";
+import heart from "@aivenio/aquarium/icons/heart";
+import trash from "@aivenio/aquarium/icons/trash";
+import loading from "@aivenio/aquarium/icons/loading";
+import { formatTimeMinutes } from "@/utils";
+import { useAppContext } from "@/context";
+import { useRecipes } from "@/hooks";
+import { RecipeCard } from "@/components/RecipeCard/RecipeCard";
+
+export default function Recipes() {
+  const { addingRecipeIdToFavorites } = useAppContext();
+  const { addRecipeToFavorites } = useRecipes();
+  const [page, setPage] = useState(1);
+
+  const {
+    data,
+    isLoading,
+    error,
+    mutate: refetchRecipes,
+  } = useSWR<RecipesResponse, ErrorResponse>(
+    `/api/recipes?page=${page}`,
+    fetcher
+  );
+
+  const rows =
+    data?.recipes.map(
+      ({ id, recipeName, rating, totalTimeMinutes, isFavorite }) => ({
+        id,
+        recipeName,
+        rating,
+        totalTime: formatTimeMinutes(totalTimeMinutes),
+        isFavorite,
+      })
+    ) ?? [];
+
+  const statistics = data?.statistics;
+
+  useEffect(() => {
+    if (data?.page) {
+      setPage(data.page);
+    }
+  }, [data?.page]);
+
+  if (error) {
+    return <EmptyState title={error.message ?? "Failed to load recipes."} />;
+  }
+
+  return (
+    <Layout>
+      <PageHeader
+        title="Statistics"
+        subtitle={
+          statistics?.recipeStatsCacheRetrievalTimeMs
+            ? `Recipe statistics cached in Aiven for Redis®, retrieved in ${statistics.recipeStatsCacheRetrievalTimeMs}ms (end-to-end with round
+          trip, results will vary depending on the region of your Aiven
+          for Redis® instance).`
+            : "Recipe statistics retrieved from PostgreSQL database. To get cached results using Aiven for Redis®, please follow the instructions to set up your Redis instance."
+        }
+      />
+      {data && (
```

Comment:
I think here we could have a placeholder for loading state similar to how the table is currently. Reason is that the whole screen jumps quite a lot every time we click to another page.

Reply:
fixed